lib/evaluate/characteristics.cc

```diff
@@ -84,7 +84,7 @@ std::optional<TypeAndShape> TypeAndShape::Characterize(
   if (auto type{DynamicType::From(object.type())}) {
     TypeAndShape result{std::move(*type)};
     result.AcquireShape(object);
-    return result;
+    return std::move(result);
```

Comment:
Isn't `std::move` unnecessary when returning a local?

Reply:
In this case, the function result type is `optional<>`, so the `move` optimizes its construction.

Reply:
However, some compilers are warning about a needless move, so I'll take it out.  Never mind.